Dockerfile

```diff
@@ -0,0 +1,55 @@
+FROM debian:bullseye AS requirements
+
+ENV CXXFLAGS="-Wno-narrowing"
+
+RUN apt-get update -qq \
+    && apt-get install -y -q --no-install-recommends \
+        build-essential=12.9 \
+        cmake=3.18.4-2+deb11u1 \
+        cmake-curses-gui=3.18.4-2+deb11u1 \
+        curl=7.74.0-1.3+deb11u1 \
+        libffi-dev=3.3-6 \
+        libpng-dev=1.6.37-3 \
+        libssl-dev=1.1.1n-0+deb11u1 \
+        python3.9=3.9.2-1 \
+        python3.9-dev=3.9.2-1 \
+        python3-pip=20.3.4-4+deb11u1 \
+        unzip=6.0-26 \
+        zlib1g-dev=1:1.2.11.dfsg-2+deb11u1 \
+    && rm -rf /var/lib/apt/lists/* /tmp/* /var/tmp/* \
+    && pip install --no-cache-dir \
```

Comment:
Would it be worth using something like poetry to install these dependencies or have it generate a `requirements.txt` file? The dependencies being installed here are different than what is currently being specified in the `pyproject.toml` and it would be good to keep those synced.

Reply:
That is probably worth doing, I didn't think of it.

Reply:
Not sure which would be easier / preferable:

1. Install poetry in docker and have it install that way.
2. Install poetry in the actions workflow, have it create a requirements.txt file that gets copied over when creating the docker container and installing via pip.

Reply:
1 seems preferable to me